### PR TITLE
Improve handling of uncollected metrics

### PR DESCRIPTION
**Pull request by wlach**
* *Don't* show them by default
* However, when they do exist and would otherwise have matched a search term, provide some helpful text explaining what's going on and a button to reveal them

### CODE REVIEW

**Comment by wlach** [comment]
Some screenshots:

image
image

**Comment by wlach** [comment]
@Iinh I think this gets around the previous pitfalls that we had around uncollected metrics (people getting lost because things expired or removed). Let me know what you think!

File: src/components/ItemList.svelte [hunk 1]
```diff
@@ -121,34 +122,51 @@
...
+        </p>
+        <button
+          class="mzp-c-button mzp-t-secondary mzp-t-md"
+          on:click={() => {
+            showUncollected = true;
+            updateURLState({ showUncollected });
+          }}>Show</button
+        >
+      {:else}
+        <p>Try entering less specific or alternate search terms.</p>
```
Comment (Iinh) [inline src/components/ItemList.svelte:166]: small nit: 
```suggestion
        <p>Try entering less specific or alternative search terms.</p>
```

**Review by Iinh — APPROVED** [review]
I like this new way of handling uncollected metrics. We decided to show all metrics by default after this issue https://github.com/mozilla/glean-dictionary/issues/561, but this looks like a much better approach. Thanks for doing this @wlach!

**Pull request merged**